### DISCUSSION TITLE
How can I set creds and stroage aliases

### DISCUSSION THREAD

**Discussion by Cnily03** [emersion/maddy#665] [Q&A] [answered]
I have read the documentation but cannot find a valid way to route some emails to one specific account (credential)

How can I implement such functions:

- alias `user1@example.com` and `user2@example.com` to cred `user`

- alias `user1@example.com` and `user2@example.com` to imapsql `user`

As I login as `user1@example.com` or `user2@example.com`, the account info are same and the table is `user`

And the IMAP storage of `user1@example.com` and `user2@example.com` are same storaged in table `user`

**Comment by foxcpp** [comment]
https://maddy.email/multiple-domains/

**Reply by Cnily03** [reply]
I mean something more complicated.
`a@example.com` and `b@example.org` are linked to the same account and map. But `ccc@example.com` and `ccc@example.org` don't make alais.

**Reply by foxcpp** [reply, marked as answer]
Any module under "Lookup tables" can be used for auth_map/storage_map/delivery_map.
Take a look at https://maddy.email/reference/table/static, https://maddy.email/reference/table/chain/ in particular.

E.g. how your example in original post could be implemented:
```
table.chain accounts_map {
  optional_step static {
    entry user1@example.com user
    entry user2@example.com user
  }
  step identity # default, ccc@* -> ccc@*
}

# Use it everywhere.
submission ... {
  auth_map &accounts_map
}
imap ... {
  auth_map &accounts_map
  storage_map &accounts_map
}
storage.imapsq local_storage {
  delivery_map &accounts_map
}
```

**Reply by stellarpower** [reply]
Is it also possible to chain for authentication providers? I have two separate ways to authenticate users:
```
auth.pass_table authentication_one {
    ...
}

auth.pass_table authentication_two {
    ...
}

imap tls://0.0.0.0:4993 tcp://0.0.0.0:4143 {
    auth &authentication_one
    auth &authentication_two
}

```

I'd like to combine them into one module I can refer to just once, e.g.:
```
table.chain registered_user {
    auth.pass_table authentication_one {
        ...
    }
    
    auth.pass_table authentication_two {
        ...
    }
}

imap tls://0.0.0.0:4993 tcp://0.0.0.0:4143 {
    auth &registered_user
}

```

Tried a couple of permutations but no luck so far, is thi possible?
Thanks
